language/vm/vm-runtime/src/move_vm.rs

```diff
@@ -72,17 +90,148 @@ impl VMExecutor for MoveVM {
             // needs to be fixed.
             VMRuntime::new(&*arena, config)
         });
-        vm.rent(|runtime| runtime.execute_block_transactions(transactions, state_view))
+
+        let mut result = vec![];
+        let blocks = chunk_block_transactions(transactions);
+        let mut data_cache = BlockDataCache::new(state_view);
+        for block in blocks {
+            match block {
+                TransactionBlock::UserTransaction(txns) => {
+                    let outs = vm.rent(|runtime| {
+                        execute_user_transaction_block(txns, runtime, &mut data_cache, state_view)
+                    });
+                    result.append(&mut outs?);
+                }
+                TransactionBlock::BlockPrologue(block_metadata) => {
+                    result.push(vm.rent(|runtime| {
+                        process_block_metadata(block_metadata, runtime, state_view, &mut data_cache)
+                    }))
+                }
+                // TODO: Implement the logic for processing writeset transactions.
+                TransactionBlock::WriteSet(_) => unimplemented!(""),
+            }
+        }
+
+        Ok(result)
+    }
+}
+
+/// Transactions divided by transaction flow.
```

Comment:
I moved this block logic here because it's the entry point.
This is likely to evolve with the new design of the Libra layer (@runtian-zhou) so I am just having this close to the entry point and for simplicity

Reply:
Shall we rename this file to `libra_vm.rs`?

Reply:
I am ok doing it and if you think it's the right thing to do I'll do it

Reply:
Go for it!